docs/proposals/20221003-In-place-propagation-of-Kubernetes-objects-only-changes.md

```diff
@@ -0,0 +1,220 @@
+
+2. Metadata from ClusterClass.Spec.Workers.MachineDeployments and Cluster.Spec.Workers.Topology.ControlPlane apply consistently to:
+- The MachineDeployment objects (both top level metadata and spec.template.metadata).
+- The InfrastructureMachineTemplate object referenced by the MachineDeployment (top level metadata only).
+- The BottstrapTemplate object referenced by the MachineDeployment (top level metadata only).
+
+Please note that above changes solves [Labels and annotations for MachineDeployments and KubeadmControlPlane created by topology controller](https://github.com/kubernetes-sigs/cluster-api/issues/7006);
+also, we are stopping to propagate existing labels on ClusterClass templates to objects created from them in favor of always
+propagating the same set of labels as described above.
+
+Note: as of today topology controllers already propagates ClusterClass and Cluster Topology metadata changes in place when possible
+in order to avoid unnecessary template rotation with the consequent Machine rollout; we do not foresee changes to this logic
+in order to support above propagation rules.
+
+#### Simplified Metadata propagation from MachineDeployment to MachineSet
+
+We are going to drop current behaviour inherithed from Kubernetes's Deployment controller, that computes top level labels in MachineSet
+using a different source from the one used for annotations. Instead we propose a more straight forward propagation:
+
+1. Top level metadata from MachineDeployment propagates to top level metadata in MachineSet.
+2. spec.template.metadata from MachineDeployment propagates to spec.template.metadata in MachineSet.
+
+In order to prevent unnecessary rollouts, the above propagation should happen in place (see [in place propagation](#in-place-propagation));
+same considerations apply to metadata propagation from MachineSet to machines.
+
+#### In Place Metadata propagation from KubeadmControlPlane to Machines
+
+In order to prevent unnecessary rollouts, metadata propagation from KubeadmControlPlane to Machines should happen in place (see [in place propagation](#in-place-propagation)).
+
+#### Label Sync Between Machine and underlying Kubernetes Nodes
+
+As discussed in [Label Sync Between Machine and underlying Kubernetes Nodes](link to the proposal as soon as it merge) we are propagating from
+the Machine to corresponding Kubernetes Node only labels with a well known prefix or a well known domain.
+
+### Propagation of fields impacting only Kubernetes objects or controller behaviour
+
+Apart from labels and annotations, there are also other fields that flow downs from ClusterClass to KubeadmControlPlane/MachineDeployments and
+ultimately to Machines.
+
+Some of them can be considered like labels and annotations, because they have impacts only only Kubernetes objects or controller behaviour, but
+not on the Machine itself. Examples are `MinReadySeconds`, `NodeDrainTimeout`, `NodeVolumeDetachTimeout`, `NodeDeletionTimeout`.
```

Comment:
Machine here as written = kubernetes CR. I think you are referring to the provider instance which remains immutable

Reply:
I will clarify this, good catch